### PR TITLE
feat(DTFS2-6892): add controllers and error handler wrapper to start routes

### PR DESCRIPTION
**Pull request by AlexBramhill**
## Introduction :pencil2:
updates start routes in portal api to use error wrapper

extracts out start routes to controllers

### CODE REVIEW

File: portal/server/controllers/start/before-you-start-bank-deal.controller.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { constructPayload, errorHref, postToApi, requestParams } from '../../helpers';
```
Comment (AlexBramhill) [inline portal/server/controllers/start/before-you-start-bank-deal.controller.js:1]: I am not adding tests as this is only moving code around -- and theres a lot to do!

File: portal/server/controllers/start/before-you-start.controller.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { errorHref, generateErrorSummary } from '../../helpers';
+import beforeYouStartValidation from '../../validation/before-you-start';
+
+export const getBeforeYouStartPage = async (req, res) => {
+  const { mandatoryCriteria } = req.apiData;
+  res.render('before-you-start/before-you-start.njk', {
```
Comment (abhi-markan) [inline portal/server/controllers/start/before-you-start.controller.js:6]: Can these be inside `template` directories to keep it consistent please.

File: portal/server/controllers/start/before-you-start.controller.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { errorHref, generateErrorSummary } from '../../helpers';
+import beforeYouStartValidation from '../../validation/before-you-start';
+
+export const getBeforeYouStartPage = async (req, res) => {
```
Comment (abhi-markan) [inline portal/server/controllers/start/before-you-start.controller.js:4]: Docstring please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: portal/server/controllers/start/before-you-start.controller.js (hunk 3 above)
Comment (AlexBramhill) [reply portal/server/controllers/start/before-you-start.controller.js:4]: sorry -- These are just me pulling out controller functions into controllers -- If this is the method i'm going to go with, there'll be a huge effort in doing this, so I'm not looking to touch the actual content if thats okay?

**Comment by AlexBramhill** [comment]
Moved to drafts due to reassessment of deliverables

**Pull request closed without merging**